telegram/ext/picklepersistence.py

```diff
@@ -0,0 +1,235 @@
+            persistence class. Default is ``True``.
+        store_chat_data (:obj:`bool`, optional): Whether user_data should be saved by this
+            persistence class. Default is ``True``.
+        single_file (:obj:`bool`, optional): When ``False`` will store 3 sperate files of
+            `filename_user_data`, `filename_chat_data` and `filename_conversations`. Default is
+            ``True``.
+        on_flush (:obj:`bool`, optional): When ``True`` will only save to file when :meth:`flush`
+            is called and keep data in memory until that happens. When False will store data on any
+            transaction. Default is ``False``.
+    """
+
+    def __init__(self, filename, store_user_data=True, store_chat_data=True, singe_file=True,
+                 on_flush=False):
+        self.filename = filename
+        self.store_user_data = store_user_data
+        self.store_chat_data = store_chat_data
+        self.single_file = singe_file
+        self.on_flush = on_flush
+        self.user_data = None
+        self.chat_data = None
+        self.conversations = None
+
+    def load_singlefile(self):
+        try:
+            filename = self.filename
+            with open(self.filename, "rb") as f:
+                all = pickle.load(f)
+                self.user_data = defaultdict(dict, all['user_data'])
+                self.chat_data = defaultdict(dict, all['chat_data'])
+                self.conversations = all['conversations']
+        except IOError:
+            self.conversations = {}
+            self.user_data = defaultdict(dict)
+            self.chat_data = defaultdict(dict)
+        except pickle.UnpicklingError:
+            raise TypeError("File {} does not contain valid pickle data".format(filename))
+        except Exception:
+            raise TypeError("Something went wrong unpickling {}".format(filename))
+
+    def load_file(self, filename):
+        try:
+            with open(filename, "rb") as f:
+                return pickle.load(f)
+        except IOError:
+            return None
+        except pickle.UnpicklingError:
+            raise TypeError("File {} does not contain valid pickle data".format(filename))
+        except Exception:
+            raise TypeError("Something went wrong unpickling {}".format(filename))
+
+    def dump_singlefile(self):
+        with open(self.filename, "wb") as f:
+            all = {'conversations': self.conversations, 'user_data': self.user_data,
+                   'chat_data': self.chat_data}
+            pickle.dump(all, f)
+
+    def dump_file(self, filename, data):
+        with open(filename, "wb") as f:
+            pickle.dump(data, f)
+
+    def get_user_data(self):
+        """Returns the user_data from the pickle file if it exsists or an empty defaultdict.
+
+        Returns:
+            :obj:`defaultdict`: The restored user data.
+        """
+        if self.user_data:
+            pass
```

Comment:
I'm not sure I quite understand these passes.. Could you explain it for me?

Reply:
What do you mean with: `aren't sure anyone can use... I` 

It will check if it has user_data loaded in memory in the `PicklePersistence`. If it does (pass) it will just return a copy of it. If not it will try to load it from a pickle.

Reply:
Riiight, pass.. not return. Got it. Brain derped a bit :)

Reply:
What do you mean with
`What do you mean with: aren't sure anyone can use... I`
Now I'm confused...

Reply:
I was looking in the wrong view and comments got messed up. NVM me. found it :D